<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="description" content="A web resource for prognostic analyses of exons across human cancers using clinical transcriptomes">
        <meta name="author" content="Gungor Budak">
        <title>Documentation &ndash; ExSurv</title>
        <link rel="shortcut icon" href="assets/img/favicon.png" type="image/png">
        <link rel="stylesheet" href="assets/css/libs/bootstrap.min.css">
        <link rel="stylesheet" href="assets/css/libs/font-awesome.min.css">
        <link rel="stylesheet" href="assets/css/main.css">
    </head>
    <body>
    <!-- Navigation -->
    <nav class="navbar sticky-top navbar-expand-md navbar-light bg-light mb-4">
      <a class="navbar-brand" href="/">ExSurv</a>
      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse" id="navbarSupportedContent">
        <ul class="navbar-nav">
            <li class="nav-item">
              <a class="nav-link" href="/">
                <!-- <i class="fa fa-home"></i> -->
                Start
              </a>
            </li>
            <li class="nav-item active">
              <a class="nav-link" href="/docs.html">
                <!-- <i class="fa fa-book"></i> -->
                Docs
              </a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="/contact.html">
                <!-- <i class="fa fa-envelope"></i> -->
                Contact
              </a>
            </li>
        </ul>
        <form class="form-inline my-2 my-lg-0" style="flex-grow: 1;">
          <div class="input-group" style="flex-grow: 1;">
            <input type="text" name="query" class="form-control form-control-sm" style="flex-grow: 1;" placeholder="Search for a gene using its symbol or Ensembl ID e.g. XIST or ENSG00000229807" title="Enter a gene symbol or an Ensembl Gene ID" value="">
            <input type="text" name="pval" class="form-control form-control-sm" style="flex: 0 1 4em;" placeholder="p-value" value="0.05" title="p-value threshold for survival analysis results">
            <select class="form-control form-control-sm custom-select custom-select-sm" style="flex: 0 1 6em;" name="cancer" title="Cancer type">
                <option value="BRCA">BRCA - Breast invasive carcinoma</option>
                <option value="GBM" selected="selected">GBM - Glioblastoma multiforme</option>
                <option value="KIRP">KIRP - Kidney renal papillary cell carcinoma</option>
                <option value="LIHC">LIHC - Liver hepatocellular carcinoma</option>
            </select>
            <select class="form-control form-control-sm custom-select custom-select-sm" style="flex: 0 1 6em;" name="classification" title="Classification method for expression values">
              <option value="Median" selected="selected">Median</option>
              <option value="Quartile">Quartile</option>
            </select>
            <div class="input-group-append">
              <button type="submit" class="btn btn-primary btn-sm button-search" title="Search for survival analysis plots for a gene">
                <i class="fa fa-search"></i>
            </button>
            </div>
          </div>
        </form>
      </div>
    </nav>
    <!-- /Navigation -->

        <div class="container">
            <h1>Documentation</h1>
            <hr>
            <div class="row">
                <div class="col-md-9">
                    <a class="anchor" name="introduction"></a>
                    <h2>Introduction</h2>
                    <p class="text-justify">
                        ExSurv is a web resource for studying the survival contributions of exons across human cancers using RNA-seq data. ExSurv is the first web server which provides exon level survival significance by using the RNA-seq expression datasets and the clinical metadata for four cancer types from The Cancer Genome Atlas (TCGA) project. We pre-calculated the prognostic significance of more than 600000 annotated exons in Ensembl using survival package in R. We stored the TCGA clinical data, exon survival p-values and the expression of the significant exons for visualizing the survival curves in a MySQL database. We developed an integrated backend using PHP and R and used JavaScript in the frontend. The PHP/R backend is reponsible for querying the MySQL database upon user input, calling R to visualize the corresponding database results (using survival package) and returning these results to the frontend. In the frontend, the results are shown to the users in an organized way as a table where each row corresponds to an exon in the queried gene symbol or Ensembl gene ID. It is possible to export the survival plots in SVG (scalable vector graphics) format and the raw data used to generate the plot in TSV (tab-separated values) format.
                    </p>
                    <a class="anchor" name="usage"></a>
                    <h2>Usage</h2>
                    <p class="text-justify">
                        Using the form in the navigation bar across this website, it is possible to query our database for survival contributions of exons given a gene symbol or Ensembl gene ID, a p-value threshold (defaulting to 0.05), a cancer type (defaulting to GBM - Glioblastoma multiforme) and a classification method (defaulting to Median). After search is done, the results (if there is any available under these conditions) are listed as a table where each row corresponds to an exon and it has exon ID, transcript ID, gene ID, gene symbol, hazard ratio, hazard ratio p-value, hazard ratio q-value, plot and export options columns. The identifiers (IDs) and the symbol are linked to their sources which help users to investigate more on these results. The plots can be zoomed in by clicking on them or can be exported as SVG. We are also providing an export option for downloading the raw dataset per exon as well as <a href="exsurv_db.sql.gz" target="_blank">the entire MySQL database is available as an SQL dump via this link (in SQL format, compressed size ~2.5GB, uncompressed size ~12.5GB)</a>. Please see Figure S1 in our publication for the database schema.
                    </p>
                    <a class="anchor" name="datasets"></a>
                    <h2>Datasets</h2>
                    <p class="text-justify">
                        We downloaded the raw TCGA RNA-seq datasets from Database of Genotypes and Phenotypes (dbGaP) for BRCA (breast invasive carcinoma), GBM (glioblastoma multiforme), KIRP (kidney renal papillary cell carcinoma) and LIHC (liver hepatocellular carcinoma). BRCA has 1040 samples, GBM has 174 samples, KIRP has 287 samples and LIHC has 368 samples. Please see Table 1 in our publication for more statistics.
                    </p>
                    <a class="anchor" name="preprocessing"></a>
                    <h2>Preprocessing</h2>
                    <p class="text-justify">
                        The raw RNA-seq reads are downloaded from dbGaP and they are aligned against human genome 38 using HISAT (Hierarchical Indexing for Spliced Alignment of Transcripts). Next, the alignment files are given to StringTie for exon expression quantification. After the expression values are obtained, we classified the exons into "High" and "Low" expressed groups using "Median" approach where the exons above the median expression value are called "High" and below ones called "Low". We also did the classification based on top and bottom quartiles and named "High" and "Low" expressed, respectively. Next, the survival analysis is done on these two different classifications separately in R (survival package) using Kaplanâ€“Meier estimator and log-rank test as well as Cox proportional hazard ratio. Since the final significance values were very similar for the two methods, we are only showing p-value and q-values (obtained using Benjamini-Hochberg correction method) from Cox proportional hazard ratio analysis. The final result tables are imported to a MySQL database.
                    </p>
                    <a class="anchor" name="web-server"></a>
                    <h2>Web server</h2>
                    <p class="text-justify">
                        The web server consists of a frontend and a backend. The frontend is interacting with user through a simple form where the user enters a gene symbol or an Ensembl gene ID, p-value threshold, a cancer type and a classification method. It also visualizes the results after the search and provides export options for obtained results. The backend is responsible for querying the database, running R to visualize query results (to obtain survival plots) and sending the results to the frontend.
                    </p>
                </div>
                <div class="col-md-3">
                    <h2>Content</h2>
                    <ul>
                        <li><a href="#introduction">Introduction</a></li>
                        <li><a href="#usage">Usage</a></li>
                        <li><a href="#datasets">Datasets</a></li>
                        <li><a href="#preprocessing">Preprocessing</a></li>
                        <li><a href="#web-server">Web server</a></li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="footer">
                <hr>
                <p class="text-muted">
                    ExSurv: A web resource for prognostic analyses of exons across human cancers using clinical transcriptomes
                </p>
            </div>
        </div>

        <script src="assets/js/libs/jquery-3.4.1.min.js" charset="utf-8"></script>
        <script src="assets/js/libs/bootstrap.bundle.min.js" charset="utf-8"></script>
        <script src="assets/js/libs/featherlight.min.js" charset="utf-8"></script>
        <script src="assets/js/libs/spin.min.js" charset="utf-8"></script>
        <script src="assets/js/libs/FileSaver.min.js" charset="utf-8"></script>
        <script src="assets/js/main.js" charset="utf-8"></script>
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

            ga('create', 'UA-48281803-14', 'auto');
            ga('send', 'pageview');
        </script>
    </body>
</html>
